kubernetes/cache/kube_cache.go

```diff
@@ -537,6 +538,38 @@ func (c *kubeCache) GetDaemonSet(namespace, name string) (*apps_v1.DaemonSet, er
 	return retDS, nil
 }
 
+func (c *kubeCache) GetDaemonSetsWithSelector(namespace string, selectorLabels map[string]string) ([]apps_v1.DaemonSet, error) {
+	defer c.cacheLock.RUnlock()
+	c.cacheLock.RLock()
+
+	daemonSets, err := c.GetDaemonSets(namespace)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Have you tested this with `accessible_namespace != **`? I don't think this will work until you update the `GetDaemonSets` method to be like the istio config list methods.

Reply:
Thanks, @nrfox, I've thought that with metav1.NamespaceAll you just got the namespaces that you had permissions to. I've made the changes and some testing.

![image](https://github.com/kiali/kiali/assets/49480155/867b0187-db9f-47b3-902e-5969dde79432)

![image](https://github.com/kiali/kiali/assets/49480155/8148fc90-e621-4882-97b2-0d87a65c587a)

Not sure where, but probably we should add to the docs that, Kiali should have access to the namespace where the ztunnel is located.

Reply:
That's a good point. Maybe the ambient docs on kiali.io?